src/pages/Home.jsx

```diff
@@ -12,233 +12,194 @@ import ConfirmModal from "../components/ConfirmModal";
-    });
+  useEffect(() => {
+    setTasks(JSON.parse(localStorage.getItem("tasks")) || []);
+    setTrash(JSON.parse(localStorage.getItem("trash")) || []);
+  }, []);
+
+  useEffect(() => {
+    localStorage.setItem("trash", JSON.stringify(trash));
+  }, [trash]);
+
+  const setIsConfirmModalToOpen = () => {
+    setIsConfirmModalOpen(true);
   };
 
-  setIsConfirmModalClose = () => {
-    this.setState((prevState) => {
-      return { isConfirmModalOpen: false };
-    });
+  const setIsConfirmModalToClose = () => {
+    setIsConfirmModalOpen(false);
   };
 
-  toggleFilterDueDate = () => {
-    this.setState((prevState) => {
-      return {
-        filter: { ...prevState.filter, dueDate: !prevState.filter.dueDate },
-      };
-    });
+  const toggleFilterDueDate = () => {
+    setFilter({ ...filter, dueDate: !filter.dueDate });
   };
 
-  setCompletedStatusFilter = (status) => {
-    this.setState((prevState) => {
-      return { filter: { ...prevState.filter, status: status } };
-    });
+  const setCompletedStatusFilter = (status) => {
+    setFilter({ ...filter, status: status });
   };
 
-  setPriorityStatusFilter = (priority) => {
-    this.setState((prevState) => {
-      return { filter: { ...prevState.filter, priority: priority } };
-    });
+  const setPriorityStatusFilter = (priority) => {
+    setFilter({ ...filter, priority: priority });
   };
 
-  setIsCompleted = (id) => {
-    const updatedTasks = this.state.tasks.map((task) => {
+  const setIsCompleted = (id) => {
+    const updatedTasks = tasks.map((task) => {
       if (task.id === id) {
         return { ...task, isCompleted: !task.isCompleted };
       }
       return task;
     });
-
-    this.setState({ tasks: updatedTasks });
+    setTasks(updatedTasks);
   };
 
-  clearAllCompletedTasks = () => {
-    let updatedList = this.state.tasks.filter(
-      (task) => task.isCompleted !== true
-    );
+  const clearAllCompletedTasks = () => {
+    let updatedList = tasks.filter((task) => task.isCompleted !== true);
 
-    this.setState((prevState) => {
-      if (JSON.stringify(prevState.tasks) === JSON.stringify(updatedList)) {
-        alert("You have no completed tasks to remove");
-      } else {
-        return { tasks: updatedList };
-      }
-    });
+    if (JSON.stringify(tasks) === JSON.stringify(updatedList)) {
+      alert("You have no completed tasks to remove");
+    } else {
+      setTasks(updatedList);
+    }
   };
 
-  setIsAddFormOpen = () => {
-    this.setState((prevState) => {
-      return { isAddFormOpen: !prevState.isAddFormOpen };
-    });
+  const toggleIsAddFormOpen = () => {
+    setIsAddFormOpen(!isAddFormOpen);
   };
 
-  addNewTask = (task) => {
-    this.setState((prevState) => {
-      return { tasks: [...prevState.tasks, task] };
-    });
+  const addNewTask = (task) => {
+    setTasks([...tasks, task]);
   };
 
-  setIsEditFormOpen = () => {
-    this.setState({
-      isEditFormOpen: true,
-    });
+  const setIsEditFormToOpen = () => {
```

Comment:
Unnecessary One liner method

Reply:
I need this state in the home component as it deals with the opening and closing of a modal upon click of a button in another component. And the setState needs to be inside a method, otherwise they cannot be passed in as props. I am not sure how to bypass this other than using state management libraries or Context. Maybe we can delay this refinement if/until we implement context in the app